src/context/watchLaterContext.jsx

```diff
@@ -0,0 +1,40 @@
+import { createContext, useContext, useState } from 'react'
+import axios from 'axios'
+import { useNavigate } from 'react-router-dom'
+import { useAuth } from 'context'
+import checkIfPresent from 'utils/functions/checkIfPresent'
+
+const watchLaterContext = createContext(null)
+
+function WatchLaterProvider( { children } ){
+
+    const [ watchLaterState, setWatchLater] = useState([])
+    const { authState: {isLoggedIn, token} } = useAuth()
+    const navigateTo = useNavigate()
+
+    function addToWatchLater( item ){
+        isLoggedIn ?
+        !checkIfPresent(item._id, watchLaterState) && setWatchLater([...watchLaterState, item]) 
+        : navigateTo('/login')
+    }
+    
+    function removeFromWatchLater( item ){
+        setWatchLater(watchLaterState.filter(video => video._id != item._id))
+    }
+
+    function clearWatchLater(){
+        isLoggedIn && setWatchLater([])
+    }

```

Comment:
Suggestion:- For now, it is working fine, because we are using a mock backend. But we are supposed to do API calls and add the data to the backend for the watch later.

Reply:
Have added api calls to post and delete the data.